odc/geo/types.py

```diff
@@ -21,8 +21,10 @@
 
 MaybeInt = Optional[int]
 MaybeFloat = Optional[float]
-Nodata = Union[float, int, str]
-MaybeNodata = Optional[Nodata]
+FillValue = Union[float, int]
+Nodata = Union[float, int, None]
+MaybeNodata = Union[float, int, str, None]
+MaybeAutoNodata = Union[float, int, str, None, Literal["auto"]]
 T = TypeVar("T")
```

Comment:
`MaybeNodata = Nodata | str` seems unexpected as `Maybe` usually means `or None`.  I understand that str might be necessary for cases where `nodata in ('nan', 'NAN', 'NaN')` but I'm not sure why this gets a "Maybe" label?

Reply:
Oh I see, there's a conversion function in `math.py`

I still think the naming here is confusing but I don't have a good suggestion for a replacement.  Maybe `ApiNodata`?  As in the nodata type the user sees in the API (as opposed the internal nodata value passed to GDAL)?

Reply:
that's more of a compatibility issue, before this `Nodata` could not be `None`, but it could be a `str`, so there was a `MaybeNodata` also. I guess we could delete it assuming nobody was using it.